### ISSUE TITLE
[a11y]: OverflowMenu: browser focus lost on tab key

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#17983]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.63

### React version

React

### Automated testing tool and ruleset

N/A

### Assistive technology

N/A

### Description

* If the focus is on a dropdown menu item, and the user hits the tab key, focus moves to the dropdown menu itself. It should go to the element after the ellipsis button for opening the overflow menu.

* If the user tabs again, focus gets lost (it goes back to the browser address bar or something like that). It should go to the element after the ellipsis button for opening the overflow menu.

`MenuButton` does better than `OverflowMenu`.  With `MenuButton`, hitting the tab key from a `MenuItem` just doesn't do anything.  Not ideal, but not too bad either.

You can compare the behavior of `MenuButton` and `OverflowMenu` at https://stackblitz.com/edit/github-ihxqde-xrt7yt?file=src%2FApp.jsx,package.json.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default

### Steps to reproduce

You can reproduce on https://react.carbondesignsystem.com/?path=/story/components-overflowmenu--default or https://stackblitz.com/edit/github-ihxqde-xrt7yt?file=src%2FApp.jsx,package.json.

Behavior varies slightly depending on `focusTrap`, see #13115, but with either setting focus eventually jumps back to the browser address bar.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alina-jacob** [comment]
Noting down some feedback from Michael G, discussed in the design crit call. 

Tab stop lands you on the overflow menu, Enter opens the menu, Arrow keys navigate it, Enter selects an option. 
After selecting an option form the overflow menu, the focus should land on the action invoked. 

**Example:** If the "Rename app" is selected, as per the general UX flow, the focus should land on some sort of text-input or in-line editing input field that allows you to rename the app. 
<br> 
![Image](https://github.com/user-attachments/assets/ae0c4558-b24a-498b-be8a-828c8d01f7ac)

Once editing/other actions are completed, the focus can either show up on the current overflow menu (as it does with the Esc key) or it can land on the previous or next Tab stop as per user requirements. 

---

**Note:** Landing the focus on current overflow menu or next Tab stop is not possible for scenarios where the last element has been deleted; leaving the component / screen / dashboard empty. In such cases the Tab stop lands on the previous or other desired stops. 
In the image below, after deleting the row, the focus should be set to land on the "selectable checkmark" on the data-table header or some other action as required. 

![Image](https://github.com/user-attachments/assets/b7e19565-ee1c-43e7-89db-a4687decfa27)

**Comment by wkeese** [comment]
Update: For MenuButton (or rather, the menu items underneath it), tab key was ignored in @carbon/react 1.63.   But in @carbon/react 1.80, the tab key jumps to the browser address bar, just like OverflowMenu.
